src/docs/releases/2.1.0.md

```diff
@@ -165,6 +164,34 @@ The `NamedRecipeStepHandler` class has been introduced to streamline the impleme
 
 A new abstract class, `DeploymentSourceBase<TStep>`, has been introduced to streamline and simplify deployment source implementations. Instead of directly implementing the `IDeploymentSource` interface, it is now recommended to inherit from `DeploymentSourceBase<TStep>`. This change reduces complexity and improves the maintainability of deployment sources.
 
+### Content Types
+
+#### New Event Handlers
+
+We've introduced the `IContentDefinitionHandler` interface, which allows you to intercept and modify content definitions as they are being built, prior to being cached. This provides finer control over the creation of content types, enabling customization at various stages of the content definition process.
+
+The following events are available for handling different aspects of content definition:
+
+- `ContentTypeBuilding(BuildingContentTypeContext context)`: Triggered during the creation of a content type.
+- `ContentTypePartBuilding(ContentTypePartContextBuilding context)`: Triggered while constructing a content type part.
+- `ContentPartFieldBuilding(ContentPartFieldContextBuilding context)`: Triggered when building fields within a content part.
+- `ContentPartDefinitionBuilding(ContentPartDefinitionContextBuilding context)`: Triggered during the setup of content part definitions.
+
+These event handlers grant you the flexibility to adjust definitions at each step, providing greater customization capabilities.
+
+#### System-Type Content Parts
```

Comment:
In the code it's called `IsRemovable`, not `IsSystem`. Keep this consistent. Also, to prevent confusion with content parts, I'd just call this "System Content Types" (as there are System Roles).

Reply:
its not just a type which is way I am trying to stay away from calling it Content Type. the same settings can be applied to part and field too.

Reply:
I wanted to say "System Content Parts".

Reply:
still the usage of Type Part or Field would be confusing. How about SystemDefined?

Reply:
"System Content Parts and Fields"? :) Just in this title. And in the code, simply `IsSystem`.

Reply:
done

Reply:
In the code it's still `IsRemovable`.

Reply:
Yes. Per @sebastienros request

Reply:
Why would we call it system in the docs (including code comments) and removable in the code?

Reply:
IsRemovable would be a behavior of the system type. maybe later we want to add another flag for a different behavior.

Reply:
Then it shouldn't be called system, but removable. The system roles are also called "system", where that also means non-removable.

Reply:
Can you be more specific on what you want changed?

Reply:
This is all:

```diff
public class ContentSettings
{
-    public bool IsRemovable { get; set; } = true;
+    public bool IsSystem { get; set; } = true;
}
```

Reply:
I changed it to `IsSystemDefined` to make it a complete name.